### PR TITLE
Domains: Add helper text in site redirect settings update dialog

### LINKED ISSUE #42317: Add a disable/enable option to site redirection subscriptions
#### Steps to reproduce

1. Create a site direction: https://wordpress.com/domains/add/site-redirect/
2. Head to "Redirect Settings" https://wordpress.com/domains/manage/mrfoxtalbot.com/redirect-settings/

#### What I expected
I expected to be able to "switch off" the redirection without having to cancel it. Being able to do this could be necessary in the context of troubleshooting, for example.

#### What happened instead
There is no option to do this, only to update the URL being redirected to.

#### Browser / OS version
MacOS 10.13.6 (17G2307)
Google Chrome 79.0.3945.88

#### Screenshot / Video
Screen Shot 2020-05-18 at 08 54 38

#### Context / Source
This was brought to my attention by @igmoweb in the context of trying to use WordPress.com with the REST API, with the front-end showing on an external site, while maintaining the RSS feed in the original wordpress.com site. 

Not being able to switch this redirection made it hard to test/troubleshoot this.

### PR DESCRIPTION
**Pull request by ncardeli**
To disable a site redirect users have to make it the *non* primary domain for the site.
Users trying to disable a site redirect, who might not know about this, may end up in the site redirect settings update dialog, trying to disable it. 

Users would benefit from more explicits ways to accomplish this task. As mentioned by @klimeryk in #42317, this could be addressed by the work @fditrapani and @hambai are doing around domain management. In the meantime, it would be helpful to put an explicit helper text in the site redirect settings update dialog with a link to the support docs, where there's a section that explains how to disable a site redirect.

#### Changes proposed in this Pull Request

* Added a helper text in the site redirect settings dialog with a localized link to the site redirect support docs.

##### Before

imagen

##### After

imagen

#### Testing instructions

* Create a site redirect (omit this step if there's one already defined)
* Go to Manage > Domains and click the domain of the site redirect
* Update the site redirect by clicking "Redirect settings"
* Verify that under the form input field there's a message that says: "All domains on this site will redirect here as long as this domain is set as your primary domain. Learn more".
* Verify that, if the interface language is English, clicking the "Learn more" link takes you to https://wordpress.com/support/site-redirect/.
* Verify that, if the interface language is other than English, clicking the "Learn more" link takes you to the localized version of https://wordpress.com/support/site-redirect/ that matches the selected interface language.

Fixes #42317

### CODE REVIEW

File: client/my-sites/domains/domain-management/site-redirect/index.jsx [hunk 1]
```diff
@@ -148,7 +149,19 @@ class SiteRedirect extends React.Component {
...
 								<p className="site-redirect__explanation">
-									{ translate( 'All domains on this site will redirect here.' ) }
+									{ translate(
+										'All domains on this site will redirect here as long as this domain is set as your primary domain. ' +
+											'{{learnMoreLink}}Learn more{{/learnMoreLink}}',
+										{
+											components: {
+												learnMoreLink: (
+													<a
+														href={ localizeUrl( 'https://wordpress.com/support/site-redirect/' ) }
```
Comment (klimeryk) [inline client/my-sites/domains/domain-management/site-redirect/index.jsx:159]: Props on finding the `localizeUrl` method - I actually did not know it was a thing 😅 👏 
But I'd love it if you could slightly tweak the `<a>` component here, to be in line with other places we've included links to support. See for example here:

https://github.com/Automattic/wp-calypso/blob/180a2d16e0f92e4980c4e2da77c16851586e16e1/client/my-sites/domains/domain-management/transfer/transfer-out/transfer-prohibited.jsx#L24

So, basically:
 - `target="_blank" rel="noopener noreferrer"` ([just to be on the safe side](https://pointjupiter.com/what-noopener-noreferrer-nofollow-explained/))
 - and moving the URL itself to https://github.com/Automattic/wp-calypso/blob/180a2d16e0f92e4980c4e2da77c16851586e16e1/client/lib/url/support.js - where I see now that we _do_ use `localizeUrl`... and the old (but working, thanks to rewrite rules, of course) URL for the root of support docs 🤦 Feel free to update it too, but probably ideally we'd do it in a new PR (which you are welcome to do as well!). The reason for putting the URLs in this one file is that it's easier to search for them and when they need updating, it's also a much smaller PR. Case in point: https://github.com/Automattic/wp-calypso/pull/44210

**Comment by klimeryk** [comment]
Thanks for submitting the PR, @ncardeli! I tested it and it's working as expected, just left one nitpick about the link 🙇

**Comment by fditrapani** [comment]
From the screenshots this looks good! It looks like you solved the Learn more breaking into two lines from here.

**Comment by ncardeli** [comment]
> From the screenshots this looks good! It looks like you solved the Learn more breaking into two lines from here.

@fditrapani Not exactly, the screenshot is a bit misleading, because it wasn't taken in full width.
This is how it looks in full width:

imagen

**Comment by fditrapani** [comment]
aah ok. Thanks for sharing that. If you add this CSS to the link: `white-space: nowrap;` it'll prevent it from breaking to two lines.

File: client/my-sites/domains/domain-management/site-redirect/index.jsx (hunk 1 above)
Comment (ncardeli) [reply client/my-sites/domains/domain-management/site-redirect/index.jsx:159]: Thanks @klimeryk, all your suggestions make perfect sense.

Just one small comment: I  verified and the URL for the root of support was updated in master 3 months ago, so don't worry about it 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4117037

Thank you @ncardeli for including a screenshot in the description! This is really helpful for our translators.

**Comment by klimeryk** [comment]
Just to close the loop - the change has now been deployed 🙌

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4117037).

**Pull request merged**